cookbooks/fb_systemd/recipes/udevd.rb

```diff
@@ -18,6 +18,12 @@
 # limitations under the License.
 #
 
+link '/sbin/udevadm' do
+  to '/bin/udevadm'
+  owner 'root'
+  group 'root'
+end
```

Comment:
Recommended changes:
1. Reverse the symlink
2. Use Relative paths so the link is meaningful outside of chroots/containers. `/bin/udevadm -> ../sbin/udevadm`
3. Adding a guard to avoid making the link if the target is a symlink (which is the normal case): `not_if { File.symlink?('/sbin/udevadm') }`
4. Change the paths below to use `/bin` directly.

Reply:
@malmond77, it's not quite as simple, on modern Fedora and CentOS the whole of `/sbin` is a symlink to `/bin`, not just the one file. It might be easier to just make this path conditional using `value_for_platform` or similar instead of mucking with symlinks.

Reply:
I second @davide125 's solution - though with a slight correction. It's not that `/sbin` and `/bin` are unified it's that `/bin`/`/usr/bin` have been unified and `/sbin`/`/usr/sbin` have been unified.

You can't reverse the symlink - as the OP points out the actual binary is in /bin - so you'd have to move the binary which seems like a bad idea. `value_for_platform` is definitely the right call here.

One other thing I'll add here is that *most* platforms have moved udevadm to bin at this point, so we should almost always be using /bin/udevadm

* Debian sid - /bin
* CentOS 7 - /bin
* Ubuntu 18 - /bin
* Ubuntu 16 - /bin

My guess is that /sbin is probably not anywhere other than CentOS6 and early fedora, though I don't have access to such systems to test on.

Reply:
Thanks for all the great feedback, suggestions. I'll take a whack at reverting the symlink addition and making the recommended changes.

Reply:
So I've gone back through the following distros/versions: 
- Fedora 24-32: All of these releases seem to have `/bin/udevadm` existing
- Ubuntu 16, 18. 20: All of these releases seem to have `/bin/udevadm` existing
- Debian 8, 9, 10: All of these releases seem to have `/bin/udevadm` existing
- CentOS 6, 7, 8: @jaymzh is correct insofar that CentOS 6 appears to be missing `/bin/udevadm` (and, instead, the bin exists at `/sbin/udevadm`). In CentOS 7 (and 8), this path exists. I was unable to get a working copy of the CentOS 5 ISO but this version reached EOL in March of 2017.

All that said, would y'all deem it acceptable to have the `value_for_platform` `default`'s key-value reflect `/bin/udevadm` and then a sole key for `'centos' => { '< 6.0' => '/sbin/udevadm'}` to cover this singular edge case?

Or would it be preferable to have each platform explicitly defined?

Reply:
Yup. That's exactly what I think we should do. Thanks for the thorough checking!

Reply:
Great; thanks for confirming, @jaymzh!

I've pushed the changes and think I've covered all the discussed cases (but its still early and I'm not fully caffeinated yet, so double-check me on this commit 😆).